code/obj/item/device/lights.dm

```diff
@@ -481,3 +481,93 @@
 		on = 1
 		set_icon_state(src.icon_on)
 		src.light.enable()
+
+/obj/item/device/light/flare // Absolutely only for use in military situations and station emergencies
+	icon = 'icons/obj/lighting.dmi'
+	icon_state = "flare"
+	var/base_state = "flare"
+	var/off_state = "flare_out"
+	var/on_state = "flare_on"
+	name = "flare"
+	desc = "A hand-held military-grade flare designed for use in varied emergencies. Not safe for human consumption."
+	w_class = W_CLASS_SMALL
+	flags = ONBELT | TABLEPASS
+	var/heated = 0
+	col_r = 0.9
+	col_g = 0.1
+	col_b = 0
+	brightness = 1
+	height = 0.75
+	var/const/expiry_time = 2700
+	var/color_name = "Red"
+	light_type = null
+	var/datum/component/loctargeting/sm_light/light_c
+
+	New()
+		..()
+		light_c = src.AddComponent(/datum/component/loctargeting/sm_light, col_r*255, col_g*200, col_b*200, 255 * brightness)
+		light_c.update(0)
+
+	proc/burst()
+		var/turf/T = get_turf(src.loc)
+		make_cleanable( /obj/decal/cleanable/generic,T)
+		make_cleanable( /obj/decal/cleanable/flare_off,T)
+		make_cleanable( /obj/decal/cleanable/redglow,T)
+		qdel(src)
+
+	proc/turnon()
+		on = 1
```

Comment:
I think it'd be cool to have it so that, when activated, the flares do like 7 burn damage.